include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
I agree it would be nice to support GCC and Clang's `byte`s too. I thought I'd just add the support for MSVC `std::byte` in this change and do them in a separate pass. This is the one I need now internally ;-).